### ISSUE TITLE
Override does not work with repository type path-mapping.

### ISSUE THREAD

**Issue by temp** [puli/issues#104]
We have a package that’s installed via composer.
It defines a resource path, and our root package overrides this.
Puli should now merge both resource dirs, and override files from the subpackage with files from the root package.
But this doesn't work for the path-mapping repository, filesystem repository works fine.

**Comment by webmozart** [comment]
cc @tgalopin

**Comment by tgalopin** [comment]
I'm trying to understand the problem and whyit wasn't catched by the tests. Can I discuss it with you on Gitter @temp ?

**Comment by tgalopin** [comment]
Can you assign me this one as well please @webmozart ?

**Comment by webmozart** [comment]
You should now be able to assign tickets yourself.

**Comment by patternseek** [comment]
I've just had a very similar issue pop-up which I'll bet is the same problem. As mentioned on Gitter if I have two or more physical paths mapped to a logical path they're not merged, the last one in puli.json wins.

puli.json excerpt:
```
    "version": "1.0",
    "name": "patternseek/agile",
    "path-mappings": {
        "/app": "res",
        "/app/config": [
            "res/config",
            "var/config"
        ],
        "/app/lib": [
            "modules/tFrame/lib",
            "modules/monitor/lib",
            "modules/core/lib",
            "modules/reCaptcha/lib",
            "modules/tOrm/lib",
            "modules/mailchimp/lib",
            "modules/bugReport/lib"
```
"puli map" output excerpt:
```
The following path mappings are currently enabled:

    Package: patternseek/agile

        Puli Path                              Real Path(s)                                      
        /app                                   res
        /app/config                            res/config, var/config
        /app/lib                               modules/tFrame/lib, modules/monitor/lib,
                                               modules/core/lib, modules/reCaptcha/lib,
                                               modules/tOrm/lib, modules/mailchimp/lib,
                                               modules/bugReport/lib
```

```
 ❯ puli ls /app/lib
bugReport
```

**Comment by tgalopin** [comment]
Thanks for your input @patternseek, I'm trying to reproduce the problem and create a test covering this case.

**Comment by tgalopin** [comment]
@patternseek @temp I have worked on this issue and it might be solved. However, I don't have the same context as you, so I might have missed something. Can you test https://github.com/puli/repository/pull/40 in local and tell me if it works? I fixed various things and added a test for your issues.

**Comment by webmozart** [comment]
Fixed in puli/repository#46
